From 6b8ce07eeb11b4cc8ead6e400277f9468f6c5bd0 Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Mon, 15 Jun 2015 18:26:37 +0800
Subject: [PATCH] MLK-11115 arm: imx: double check m4 is enabled

Make a double check that m4 is enabled and run on imx7d

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
---
 arch/arm/mach-imx/src.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/src.c b/arch/arm/mach-imx/src.c
index a0d5f10..2524325 100644
--- a/arch/arm/mach-imx/src.c
+++ b/arch/arm/mach-imx/src.c
@@ -170,7 +170,7 @@ void __init imx_src_init(void)
 
 	if (cpu_is_imx7d()) {
 		val = readl_relaxed(src_base + SRC_M4RCR);
-		if ((val & BIT(3)) == BIT(3))
+		if (((val & BIT(3)) == BIT(3)) && !(val & BIT(0)))
 			m4_is_enabled = true;
 		else
 			m4_is_enabled = false;
-- 
1.8.0

